plugin/exec.go

```diff
@@ -27,15 +33,50 @@ func ExecWithOptions(opts ExecOptions) error {
 	}
 	DEBUG("Executing '%s' with arguments %v", cmdArgs[0], cmdArgs[1:])
 
+	//Encryption data is passed from the shield-pipe on fd 3
+	var encStream, decStream cipher.Stream
+	var data map[string]interface{}
```

Comment:
Use an anonymous structure for this instead of maps.

```
var data struct {
  EncryptionKey string `json:"encKey"`
  EncrypyionIV string `json:"encIV"`
}
```

Reply:
you may also want to snakify the case on those JSON tags, `enc_key`; don't forget to update the shield-pipe that sends the JSON.